service/evaluation/evaluation.go

```diff
@@ -526,38 +636,37 @@ func (svc *Service) evaluateControl(toe *orchestrator.TargetOfEvaluation, catego
 	}
 
 	// Create evaluation result
-	evalResult = &evaluation.EvaluationResult{
+	result = &evaluation.EvaluationResult{
 		Id:                         uuid.NewString(),
 		Timestamp:                  timestamppb.Now(),
-		ControlCategoryName:        categoryName,
-		ControlId:                  controlId,
+		ControlCategoryName:        control.CategoryName,
```

Comment:
Why don't you use the GetX() functions here?

Reply:
as a general rule I would say that its always faster to directly access the property instead of the getter? Accessing via the getter is only convinient when accessing optional fields which could potentially be nil.